### PR TITLE
Tests for http2 host constraint server

### PR DESCRIPTION
**Pull request by luisorbaiceta**
Add tests for secure http2 server routes with **host constraints**. This is part of the fix to the `find-my-way` package as discussed in #3408

Depends on delvedor/find-my-way#214

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by matthyk** [comment]
This should be closed since https://github.com/delvedor/find-my-way/pull/214 includes tests for this case.

**Comment by luisorbaiceta** [comment]
Isn't it worthy to add to http2 tests as an extra check for the `:authority` header? @matthyk

**Comment by matthyk** [comment]
Yes, I got that wrong. I first thought that constraints would be better tested directly in the find-my-way module, but of course, we also test the constraints using Fastify.

**Comment by jsumners** [comment]
CI is failing.

**Comment by luisorbaiceta** [comment]
As I pointed out in the description, this PR depends on another PR in the `find-my-way` repo @jsumners

**Comment by jsumners** [comment]
> As I pointed out in the description, this PR depends on another PR in the `find-my-way` repo @jsumners

Ah, I missed that. Sorry.

File: package.json [hunk 1]
```diff
@@ -179,7 +179,7 @@
     "fast-json-stringify": "^2.5.2",
     "fastify-error": "^0.3.0",
     "fastify-warning": "^0.2.0",
-    "find-my-way": "^4.1.0",
+    "find-my-way": "^5.1.0",
```
Comment (Eomm) [inline package.json:182]: We can't land this change to `main`.

We should ask @delvedor to backport the fix to 4.x or targeting the `next` branch

Comment (mcollina) [reply package.json:182]: > We should ask @delvedor to backport the fix to 4.x or targeting the next branch

@Eomm @luisorbaiceta would you like to open a PR applying the fixes in https://github.com/delvedor/find-my-way/pull/214 to https://github.com/delvedor/find-my-way/tree/4.x? You can likely cherry-pick the commit

Comment (luisorbaiceta) [reply package.json:182]: Sure! @mcollina

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**